src/frontend/hooks/useMapStyles.ts

```diff
@@ -0,0 +1,129 @@
+import * as React from "react";
+import { useFocusEffect } from "@react-navigation/native";
+import { defineMessages, useIntl } from "react-intl";
+
+import api from "../api";
+import SettingsContext from "../context/SettingsContext";
+import { MapServerStyleInfo } from "../sharedTypes";
+import { useExperiments } from "./useExperiments";
+import { useMapAvailability, onlineStyleURL } from "./useMapAvailability";
+
+// Randomly generated, but should not change, since this is stored in settings
+// if the user selects one of these "legacy" map styles
+export const DEFAULT_MAP_ID = "487x2pc8ws801avhs5hw58qnxc" as const;
+export const CUSTOM_MAP_ID = "vg4ft8yvzwfedzgz1dz7ntneb8" as const;
+
+type Status = "loading" | "error" | "success";
+type StylesAtLeastOne = [MapServerStyleInfo, ...MapServerStyleInfo[]];
+
+const m = defineMessages({
+  defaultBackgroundMapName: {
+    id: "hooks.useMapStyles.defaultMapName",
+    defaultMessage: "Default",
+    description: "The name of the default background map",
+  },
+  offlineBackgroundMapName: {
+    id: "hooks.useMapStyles.offlineMapName",
+    defaultMessage: "Offline Map",
+    description: "The name of the legacy offline background map",
+  },
+});
+
+/**
+ * Returns list of available map styles. If the backgroundMaps experiment is not
+ * enabled then this is always a single item - either the default online map or
+ * the custom map is available
+ */
+export function useMapStyles() {
+  const { formatMessage: t } = useIntl();
+  const customStyleAvailability = useMapAvailability("custom");
+  const [{ backgroundMaps }] = useExperiments();
+  const [settings, setSettings] = React.useContext(SettingsContext);
+  const [stylesList, setStylesList] = React.useState<MapServerStyleInfo[]>([]);
+  // There is nothing to load if background maps experiment is not enabled
+  const [status, setStatus] = React.useState<Status>(
+    backgroundMaps ? "loading" : "success"
+  );
+
+  useFocusEffect(
+    React.useCallback(() => {
+      if (!backgroundMaps) return;
+      let didCancel = false;
+      api.maps
+        .getStyleList()
+        .then(styles => {
+          if (didCancel) return;
+          setStylesList(styles);
+          setStatus("success");
+        })
+        .catch(() => didCancel || setStatus("error"));
+
+      return () => {
+        didCancel = true;
+      };
+    }, [backgroundMaps])
+  );
+
+  const defaultStyle = React.useMemo(() => {
```

Comment:
`useMemo` isn't really needed here because these are not expensive calculations and we are not doing any equality tests on the resulting object (the `styles` array is a new object every render anyway).

Reply:
addressed via c25d4b8d5c4fb685cab2d3d81bb9bda9e9001ec9